source/NVDAObjects/UIA/wordDocument.py

```diff
@@ -68,24 +70,37 @@ def getCommentInfoFromPosition(position):
 		UIAElement=UIAElementArray.getElement(index)
 		UIAElement=UIAElement.buildUpdatedCache(UIAHandler.handler.baseCacheRequest)
 		obj=UIA(UIAElement=UIAElement)
-		if not obj.parent or obj.parent.name!='Comment':
+		if not obj.parent or (
+			obj.parent.role != controlTypes.ROLE_GROUPING and not
+			obj.parent.UIAElement.getCurrentPropertyValue(UIAHandler.UIA_IsAnnotationPatternAvailablePropertyId)

```

Comment:
Please add a comment above each line answering "why?" 
It seems you have mostly answered this in response to @leonardder, putting it next to the code will likely save the next developer significant time.

However, from the comment it is not clear to me why the objects parent role must be group.

Reply:
@feerrenrut It checks for grouping because from the log viewer I see that the role of that object is grouping.
Are such comments ok?
			# Because the name of this object is language sensitive check for its role instead.
obj.parent.role != controlTypes.ROLE_GROUPING
			# And to make sure it's the right object check if it has Annotation Pattern
and not obj.parent.UIAElement.getCurrentPropertyValue(UIAHandler.UIA_IsAnnotationPatternAvailablePropertyId)

Reply:
Do you know for sure that all comment objects have a parent with role grouping? Could you theoretically remove this check and just check for `not obj.parent.UIAElement.getCurrentPropertyValue(UIAHandler.UIA_IsAnnotationPatternAvailablePropertyId` if not, why not?

It is also acceptable to keep the role_grouping check for performance reasons. For instance, if we knew that when there is no role_grouping then there will definitely not be `obj.parent.UIAElement.getCurrentPropertyValue(UIAHandler.UIA_IsAnnotationPatternAvailablePropertyId` then we can use `role_grouping` as a shortcut to avoid a UIA call.

Reply:
I removed role check and it still works. I doubt that role is connected with UIA pattern so maybe the best way here is to check only for annotation pattern.

Reply:
I'd like to get advice from @michaelDCurran here. Avoiding UIA calls where possible helps performance, but I don't want that to come at a cost to accuracy.

Reply:
For now I deleted role check.